# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: 868577854548
# Full Hash: 868577854548c7442c15ab9b1ad1213af187e01e
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2020-11-17 21:55:29
# Regressor Bug: 1671369
# File Overlap Count: 1
# Description:
#   Bug 1671369 - Change GetUsageForClient to return a UsageInfo. r=dom-workers-and-storage-reviewers,ttung
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D93777
# ==============================================================================

diff -r 5f347d3846aa -r 868577854548 dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Tue Nov 17 12:13:05 2020 +0000
+++ b/dom/quota/ActorsParent.cpp	Tue Nov 17 13:17:24 2020 +0000
@@ -1013,7 +1013,7 @@
 
   void LockedResetUsageForClient(Client::Type aClientType);
 
-  bool LockedGetUsageForClient(Client::Type aClientType, uint64_t& aUsage);
+  UsageInfo LockedGetUsageForClient(Client::Type aClientType);
 
   void LockedUpdateAccessTime(int64_t aAccessTime) {
     AssertCurrentThreadOwnsQuotaMutex();
@@ -4247,10 +4247,9 @@
   }
 }
 
-bool QuotaManager::GetUsageForClient(PersistenceType aPersistenceType,
-                                     const GroupAndOrigin& aGroupAndOrigin,
-                                     Client::Type aClientType,
-                                     uint64_t& aUsage) {
+UsageInfo QuotaManager::GetUsageForClient(PersistenceType aPersistenceType,
+                                          const GroupAndOrigin& aGroupAndOrigin,
+                                          Client::Type aClientType) {
   MOZ_ASSERT(!NS_IsMainThread());
   MOZ_ASSERT(aPersistenceType != PERSISTENCE_TYPE_PERSISTENT);
 
@@ -4258,21 +4257,21 @@
 
   GroupInfoPair* pair;
   if (!mGroupInfoPairs.Get(aGroupAndOrigin.mGroup, &pair)) {
-    return false;
+    return UsageInfo{};
   }
 
   RefPtr<GroupInfo> groupInfo = pair->LockedGetGroupInfo(aPersistenceType);
   if (!groupInfo) {
-    return false;
+    return UsageInfo{};
   }
 
   RefPtr<OriginInfo> originInfo =
       groupInfo->LockedGetOriginInfo(aGroupAndOrigin.mOrigin);
   if (!originInfo) {
-    return false;
-  }
-
-  return originInfo->LockedGetUsageForClient(aClientType, aUsage);
+    return UsageInfo{};
+  }
+
+  return originInfo->LockedGetUsageForClient(aClientType);
 }
 
 void QuotaManager::UpdateOriginAccessTime(
@@ -7970,18 +7969,16 @@
   quotaManager->mTemporaryStorageUsage -= size;
 }
 
-bool OriginInfo::LockedGetUsageForClient(Client::Type aClientType,
-                                         uint64_t& aUsage) {
+UsageInfo OriginInfo::LockedGetUsageForClient(Client::Type aClientType) {
   AssertCurrentThreadOwnsQuotaMutex();
 
-  Maybe<uint64_t>& clientUsage = mClientUsages[aClientType];
-
-  if (clientUsage.isNothing()) {
-    return false;
-  }
-
-  aUsage = clientUsage.value();
-  return true;
+  // The current implementation of this method only supports DOMCACHE and LS,
+  // which only use DatabaseUsage. If this assertion is lifted, the logic below
+  // must be adapted.
+  MOZ_ASSERT(aClientType == Client::Type::DOMCACHE ||
+             aClientType == Client::Type::LS);
+
+  return UsageInfo{DatabaseUsageType{mClientUsages[aClientType]}};
 }
 
 void OriginInfo::LockedPersist() {